CIP-1775/README.md

```diff
@@ -0,0 +1,108 @@
+---
+CIP: 1775
+Title: Delegated Spending Authorizations
+Authors: Ivan Mikushin <sorrel_cantos.0@icloud.com>
+Comments-URI:
+Status: Proposed
+Type: Standards Track
+Created: 2022-11-17
+License: CC-BY-4.0
+---
+
+# Delegated Spending Authorizations
+
+"Give me custody, or give me death!" - [@ArmySpies](https://twitter.com/ArmySpies/status/1592586212816355328)
+
+## Abstract
+
+We propose to allow users to delegate authorized spending of outputs in their wallets to smart contracts or other users, without sending said outputs out of their wallets.
+
+## Motivation
+
+In order to use any dApp on Cardano (e.g. a decentralized exchange or an NFT marketplace), a user has to **deposit tokens** to a script address, thereby **giving up custody** of the tokens.
+
+This can be problematic for a number of reasons.
+
+- The user loses control of the assets they send out to a smart-contract, unless the smart contract is specifically designed to allow for such control (e.g. giving a "Cancel trade" or "De-list NFT" button in the UI).
+- Assets sent out from a user's wallet:
+  - are no longer  **delegated** to the stake pool of the user's choice.
+  - cannot participate in **governance voting** on behalf of the user.
+- Any output, accidentally sent to a script address without a datum is effectively lost now (because of how transaction validation scripts are invoked).
+- Requiring users to deposit assets into a dapp may subject the dapp's owners to regulatory scrutiny.
+
+We believe, this is not strictly necessary and would like a way to allow users to delegate spending authorization for some outputs **within** their wallets to other users (smart contracts, multisigs, wallet addresses).
+
+### Use Cases
+
+Instead of sending tokens to the dapp smart-contract, the user creates an output in their own wallet that delegates spending authorization to a smart-contract (or other entity).
+
+Some **existing** use cases, where this can be beneficial:
+
+1. Swapping tokens on a decentralized exchange.
+2. Providing liquidity on a decentralized exchange.
+3. Listing an NFT on a marketplace.
+
+Some **New** use cases, enabled by this proposal:
+
+1. On-chain noncustodial asset management. A user can delegate managing (some of) their assets to a manager, another individual user or organization.
+2. Concurrent use of dapps. For example, a user can provide liquidity and, at the same time, have a trade open at a (different) DEX, with the same output. When the trade is executed, it looks just as though the user has withdrawn the output from the liquidity pool.
+3. Seamless interaction with L2 (sidechains or Hydra). The user doesn't need to send assets to the L2 entrypoint script/multisig, nor do they need to explicitly take them out.
```

Comment:
There's ways to achieve this with off-chain code. You can monitor a given multi-sig address for receipt of assets and mint duplicate assets inside an already running hydra head. This way, you can keep the hydra head running and doing work without explicitly starting or stopping the head. The off-chain code can operate as the on/off ramp for the head and handle continuous settlement with the main chain.

Reply:
The point is to **not** make users to send to a multisig. And, symmetrically, to not make their recipients pull the assets back (when the interaction is over).

Reply:
This seems like exactly a case where the application _needs_ exclusivity! What you're suggesting would allow me to send some funds to a Hydra head, perform transactions on there that transfer that money to someone else... and then _also_ spend the money on the mainchain. I'm not sure what would happen there but probably we'd have to cancel the whole head? Worst case it lets you steal money. This does _not_ seem desirable.

Reply:
@michaelpj It really depends on how the L2 treats concurrency. In Optimistic Concurrency control, this works. Where exclusive locking is needed, depositing to smart contracts is still available 🙂.